Travis builds failing with Python2 on OSX

See error below.

```
$ pyenv install $PYTHON
python-build: use openssl 1.1 from homebrew
python-build: use readline from homebrew
Downloading Python-2.7.12.tar.xz...
-> https://www.python.org/ftp/python/2.7.12/Python-2.7.12.tar.xz
Installing Python-2.7.12...
python-build: use readline from homebrew
ERROR: The Python ssl extension was not compiled. Missing the OpenSSL lib?
Please consult to the Wiki page to fix the problem.
https://github.com/pyenv/pyenv/wiki/Common-build-problems
BUILD FAILED (OS X 10.13.3 using python-build 20180424)
Inspect or clean up the working tree at /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/python-build.20190407210012.86694
Results logged to /var/folders/nz/vv4_9tw56nv9k3tkvyszvwg80000gn/T/python-build.20190407210012.86694.log
Last 10 log lines:
rm -f /Users/travis/.pyenv/versions/2.7.12/share/man/man1/python.1
(cd /Users/travis/.pyenv/versions/2.7.12/share/man/man1; ln -s python2.1 python.1)
if test "xno" != "xno"  ; then \
		case no in \
			upgrade) ensurepip="--upgrade" ;; \
			install|*) ensurepip="" ;; \
		esac; \
		 ./python.exe -E -m ensurepip \
			$ensurepip --root=/ ; \
	fi
The command "pyenv install $PYTHON" failed and exited with 1 during .
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,6 +15,7 @@ osx-setup-steps: &osx-setup-steps
   before_install:
     - brew update
     - brew outdated pyenv || brew upgrade pyenv
+    - brew install readline openssl xz
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -15,9 +15,12 @@ osx-setup-steps: &osx-setup-steps
   before_install:
     - brew update
     - brew outdated pyenv || brew upgrade pyenv
-    - brew install readline openssl xz
-    - export CFLAGS="-I$(brew --prefix openssl)/include"
-    - export LDFLAGS="-L$(brew --prefix openssl)/lib"
+    - brew unlink pyenv
+    - brew install pyenv@1.2.9
+      #- brew install readline openssl xz
+      #- export CFLAGS="-I$(brew --prefix openssl)/include"
+      #- export LDFLAGS="-L$(brew --prefix openssl)/lib"
+    - brew link pyenv@1.2.9
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,6 +14,7 @@ osx-setup-steps: &osx-setup-steps
   language: generic  # 'language: python' is not yet supported on macOS
   before_install:
     - brew update
+    - brew unlink pyenv
     - brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/922f1a6709446cb1c6cfb954651275bf4033eef8/Formula/pyenv.rb
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,6 +14,7 @@ osx-setup-steps: &osx-setup-steps
   language: generic  # 'language: python' is not yet supported on macOS
   before_install:
     - brew update
+    - brew outdated pyenv || brew upgrade pyenv
     - brew unlink pyenv
     - brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/922f1a6709446cb1c6cfb954651275bf4033eef8/Formula/pyenv.rb
     - if [[ "$DIRECT_STATE" == "direct" ]]; then

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,11 +16,10 @@ osx-setup-steps: &osx-setup-steps
     - brew update
     - brew outdated pyenv || brew upgrade pyenv
     - brew unlink pyenv
-    - brew install pyenv@1.2.9
+    - brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/922f1a6709446cb1c6cfb954651275bf4033eef8/Formula/pyenv.rb
       #- brew install readline openssl xz
       #- export CFLAGS="-I$(brew --prefix openssl)/include"
       #- export LDFLAGS="-L$(brew --prefix openssl)/lib"
-    - brew link pyenv@1.2.9
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,8 +16,8 @@ osx-setup-steps: &osx-setup-steps
     - brew update
     - brew outdated pyenv || brew upgrade pyenv
     - brew install readline openssl xz
-    - export CFLAGS="-I$(brew --prefix openssl)/include" \
-    - export LDFLAGS="-L$(brew --prefix openssl)/lib" \
+    - export CFLAGS="-I$(brew --prefix openssl)/include"
+    - export LDFLAGS="-L$(brew --prefix openssl)/lib"
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,7 +16,8 @@ osx-setup-steps: &osx-setup-steps
     - brew update
     - brew outdated pyenv || brew upgrade pyenv
     - brew install readline openssl xz
-    - alias pyenv='CFLAGS="-I$(brew --prefix openssl)/include" LDFLAGS="-L$(brew --prefix openssl)/lib" pyenv'
+    - export CFLAGS="-I$(brew --prefix openssl)/include" \
+    - export LDFLAGS="-L$(brew --prefix openssl)/lib" \
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -14,12 +14,7 @@ osx-setup-steps: &osx-setup-steps
   language: generic  # 'language: python' is not yet supported on macOS
   before_install:
     - brew update
-    - brew outdated pyenv || brew upgrade pyenv
-    - brew unlink pyenv
     - brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/922f1a6709446cb1c6cfb954651275bf4033eef8/Formula/pyenv.rb
-      #- brew install readline openssl xz
-      #- export CFLAGS="-I$(brew --prefix openssl)/include"
-      #- export LDFLAGS="-L$(brew --prefix openssl)/lib"
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

--- a/.travis.yml
+++ b/.travis.yml
@@ -16,6 +16,7 @@ osx-setup-steps: &osx-setup-steps
     - brew update
     - brew outdated pyenv || brew upgrade pyenv
     - brew install readline openssl xz
+    - alias pyenv='CFLAGS="-I$(brew --prefix openssl)/include" LDFLAGS="-L$(brew --prefix openssl)/lib" pyenv'
     - if [[ "$DIRECT_STATE" == "direct" ]]; then
       brew install gcc ;
       fi

